packages/ui/client/composables/client/index.ts

```diff
@@ -2,33 +2,79 @@ import { createClient, getTasks } from '@vitest/ws-client'
 import type { WebSocketStatus } from '@vueuse/core'
 import type { ErrorWithDiff, File, ResolvedConfig } from 'vitest'
 import type { Ref } from 'vue'
-import { reactive } from 'vue'
+import { reactive as reactiveVue } from 'vue'
 import { createFileTask } from '@vitest/runner/utils'
 import type { BrowserRunnerState, RunState } from '../../../types'
 import { ENTRY_URL, isReport } from '../../constants'
 import { parseError } from '../error'
 import { activeFileId } from '../params'
 import { createStaticClient } from './static'
+import type { UIFile } from '~/composables/client/types'
+import { files, testStatus } from '~/composables/tests-status'
 
 export { ENTRY_URL, PORT, HOST, isReport } from '../../constants'
 
 export const testRunState: Ref<RunState> = ref('idle')
 export const unhandledErrors: Ref<ErrorWithDiff[]> = ref([])
 
+export { files }
+
 export const client = (function createVitestClient() {
   if (isReport) {
     return createStaticClient()
   }
   else {
+    // const callbacks = /* #__PURE__ */ new WeakMap<{ key: string }, () => void>()
+    function onCollected(collectedFiles: File[] = [], cb?: () => void) {
+      try {
+        // trigger update to all files
+        collectedFiles.forEach((file) => {
+          const entry = files.value.find(f => f.id === file.id)
+          if (entry) {
+            // callbacks.get({ key: file.id })?.()
+            /* const number = */requestAnimationFrame(() => {
```

Comment:
Maybe I'm missing some cleanup, but you're not properly tearing down the `requestAnimationFrame` anymore, yeah? Shouldn't we be using `useRequestAnimationFrame` or something from VueUse to ensure we're not leaking?